carla/recourse_methods/processing/counterfactuals.py

```diff
@@ -94,3 +94,53 @@ def reconstruct_encoding_constraints(
             x_enc[:, pair[0] + argmin] = 0
 
     return x_enc
+
+
+def check_hyperparams(hyperparams: Dict, default: Dict) -> Dict:
```

Comment:
Nice work!
Suggestions:
- We could make the naming of the function and variables more generic since this function applies to any two nested dicts?
- E.g. `merge_default_parameters`?

Reply:
Sounds much better, will change this before merge 👍